plugins/slack/plugin.go

```diff
@@ -0,0 +1,155 @@
+package slack
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/checkmarx/2ms/plugins"
+	"github.com/rs/zerolog/log"
+	"github.com/slack-go/slack"
+	"github.com/spf13/cobra"
+)
+
+const (
+	slackTokenFlag            = "token"
+	slackTeamFlag             = "team"
+	slackChannelFlag          = "channel"
+	slackBackwardDurationFlag = "duration"
+	slackMessagesCountFlag    = "messages-count"
+)
+
+const slackDefaultDateFrom = time.Hour * 24 * 14
+
+type SlackPlugin struct {
+	plugins.Plugin
+	plugins.Channels
+	Token string
+}
+
+func (p *SlackPlugin) GetName() string {
+	return "slack"
+}
+
+var (
+	tokenArg            string
+	teamArg             string
+	channelsArg         []string
+	backwardDurationArg time.Duration
+	messagesCountArg    int
+)
+
+func (p *SlackPlugin) DefineCommand(channels plugins.Channels) (*cobra.Command, error) {
+	p.Channels = channels
+
+	command := &cobra.Command{
+		Use:   fmt.Sprintf("%s --%s TOKEN --%s TEAM", p.GetName(), slackTokenFlag, slackTeamFlag),
+		Short: "Scan Slack workspace",
+		Long:  "Scan Slack workspace for sensitive information.",
+		Run: func(cmd *cobra.Command, args []string) {
+			p.getItems()
+		},
+	}
+
+	command.Flags().StringVar(&tokenArg, slackTokenFlag, "", "Slack token [required]")
+	err := command.MarkFlagRequired(slackTokenFlag)
+	if err != nil {
+		return nil, fmt.Errorf("error while marking flag %s as required: %w", slackTokenFlag, err)
+	}
+	command.Flags().StringVar(&teamArg, slackTeamFlag, "", "Slack team name or ID [required]")
+	err = command.MarkFlagRequired(slackTeamFlag)
+	if err != nil {
+		return nil, fmt.Errorf("error while marking flag %s as required: %w", slackTeamFlag, err)
+	}
+	command.Flags().StringArrayVar(&channelsArg, slackChannelFlag, []string{}, "Slack channels to scan")
+	command.Flags().DurationVar(&backwardDurationArg, slackBackwardDurationFlag, slackDefaultDateFrom, "Slack backward duration for messages (ex: 24h, 7d, 1M, 1y)")
+	command.Flags().IntVar(&messagesCountArg, slackMessagesCountFlag, 0, "Slack messages count")
```

Comment:
Note I used `ArgVar` to bind the argument with its value variable, so I don't need to assign them later in the "Initialize" phase.